client/verta/verta/_tracking/experimentrun.py

```diff
@@ -1749,21 +1860,84 @@ def log_requirements(self, requirements, overwrite=False):
 
         _pip_requirements_utils.process_requirements(requirements)
 
         if self._conf.debug:
             print("[DEBUG] requirements are:")
             print(requirements)
+        python_env = Python(copy.copy(requirements))
+        requirements = six.BytesIO(six.ensure_binary(
+            '\n'.join(requirements)))  # as file-like
+        self._log_artifact("requirements.txt", requirements,
+                           _CommonCommonService.ArtifactTypeEnum.BLOB, 'txt', overwrite=overwrite)
+        try:
+            self.log_environment(python_env, overwrite=overwrite)
+        except ValueError as ve:
+            six.raise_from(ve, None)
+        except TypeError as te:
+            six.raise_from(te, None)
+        except requests.HTTPError as he:
+            pass  # Note: HTTP errors are silently ignored for compatibility with older services
+
+    def log_environment(self, env, overwrite=False):
+        """
+        Logs a Python environment to this Experiment Run.
+
+        Parameters
+        ----------
+        env : :class:`~verta.environment.Python`
+            Environment to log.
+        overwrite : bool, default False
+            Whether to allow overwriting an existing artifact with key `key`.
+
+        """
+        if not isinstance(env, Python):
+            raise TypeError(
+                "`env` must be of type Python, not {}".format(type(env)))
+
+        if self.has_environment and not overwrite:
+            raise ValueError(
+                "environment already exists; consider setting overwrite=True")
+
+        # need to call .environment on the proto object because ._as_proto produces
+        # ai.verta.modeldb.versioning.Blob, not ai.verta.modeldb.versioning.EnvironmentBlob
+        msg = _ExperimentRunService.LogEnvironment(
+            id=self.id, environment=env._as_proto().environment)
+        data = _utils.proto_to_json(msg)
+        response = _utils.make_request("POST",
+                                       "{}://{}/api/v1/modeldb/experiment-run/logEnvironment".format(
+                                           self._conn.scheme, self._conn.socket),
+                                       self._conn, json=data)
+        if response.ok:
+            # self._refresh_cache()
+            self._fetch_with_no_cache()
+        else:
+            _utils.raise_for_http_error(response)
+
+    def get_environment(self):
+        """
+        Gets the environment of this Experiment Run.
 
-        requirements = six.BytesIO(six.ensure_binary('\n'.join(requirements)))  # as file-like
-        self._log_artifact("requirements.txt", requirements, _CommonCommonService.ArtifactTypeEnum.BLOB, 'txt', overwrite=overwrite)
+        Returns
+        -------
+        :class:`~verta.environment.Python`
+            Environment of this ExperimentRun.
+
+        """
+        self._refresh_cache()
+        if not self.has_environment:
+            raise RuntimeError("environment was not previously set.")
```

Comment:
Style for error messages is to leave off ending punctuation (consistent with built-in exceptions)
```suggestion
            raise RuntimeError("environment was not previously set")
```